code/controllers/subsystem/research.dm

```diff
@@ -68,17 +79,26 @@ SUBSYSTEM_DEF(research)
 				bitcoins = single_server_income.Copy()
 				break			//Just need one to work.
 	var/income_time_difference = world.time - last_income
-
-	science_tech.last_bitcoins = bitcoins  // Doesn't take tick drift into account
-	bos_tech.last_bitcoins = bitcoins
-	unknown_tech.last_bitcoins = bitcoins
+	if(multiserver_calculation)
+		science_tech.last_bitcoins = sciencecache  // Doesn't take tick drift into account
+		bos_tech.last_bitcoins = boscache
+		unknown_tech.last_bitcoins = unknowncache
+	else
+		science_tech.last_bitcoins(bitcoins)
+		bos_tech.last_bitcoins(bitcoins)
+		unknown_tech.last_bitcoins(bitcoins)
 
 	for(var/i in bitcoins)
 		bitcoins[i] *= income_time_difference / 10
 
-	science_tech.add_point_list(bitcoins)
-	bos_tech.add_point_list(bitcoins)
-	unknown_tech.add_point_list(bitcoins) //tbh these guys can get a fuckton of points, because it isn't even being used
+	if(multiserver_calculation)
+		science_tech.add_point_list(sciencecache)
+		bos_tech.add_point_list(boscache)
+		unknown_tech.add_point_list(unknowncache) //tbh these guys can get a fuckton of points, because it isn't even being used
+	else
+		science_tech.add_point_list(bitcoins)
+		bos_tech.add_point_list(bitcoins)
+		unknown_tech.add_point_list(bitcoins)
```

Comment:
All of the above is horrible and has to be reworked unless you people want this monstrosity to be displayed for the world to see how bad web editor code can be made.